modules/useQueries.js

```diff
@@ -26,25 +28,40 @@ function useQueries(createHistory) {
       parseQueryString = defaultParseQueryString
 
     function addQuery(location) {
-      if (location.query == null)
-        location.query = parseQueryString(location.search.substring(1))
+      if (location.query == null) {
+        const { search } = location
+        location.query = parseQueryString(search.substring(1))
+        location[SEARCH_BASE_KEY] = { search, searchBase: '' }
```

Comment:
The `$searchBase` is quite a bit of a hack. The way this works is that if you have a user-pushed location that looks like e.g.

```js
{
  path: '/the/path',
  search: '?a=one',
  query: { the: 'query value' },
}
```

We end up getting producing something that looks like:

```js
{
  path: '/the/path',
  search: '?a=one&the=query+value',
  query: { the: 'query value' },
  '$searchBase': {
    searchBase: '?a=one',
    search: '?a=one&the=query+value',
  }
}
```

Suppose we want to follow the pattern above and do something like:

```js
history.push({
  ...location,
  query: { other: 'query value' },
})
```

The object that actually gets pushed looks like:

```js
{
  path: '/the/path',
  search: '?a=one&the=query+value',
  '$searchBase': {
    searchBase: '?a=one',
    search: '?a=one&the=query+value',
  }
  query: { other: 'query value' },
}
```

Since the `search` matches `history['$searchBase'].search`, we replace it with `history['$searchBase'].searchBase`, and in a roundabout way end up with `search` as `?a=one&other=query+value`, which (given the current append semantics) is what we expect.

On the other hand, if the user does something like:

```js
history.push({
  ...location,
  search: '?b=two',
  query: { other: 'query value' },
})
```

Then we end up with search as `?b=two&other=query+value`.

There are edge cases here, but I expect that people are very unlikely to run into them in practice. I intend this as a hack that we remove ASAP, but one that is necessary if we don't want to break backward compat per the current "when there is already an existing search" test case.

The idea would be something like releasing this as v1.14.0 and taking a bit more time to gather stuff like the potential `query-string` for `qs` replacement before cutting v2.x. The `TODO` below is sort of in lieu of further documentation, because I didn't want to imply that this was deservedly complicated code that we should keep - it isn't anything of the sort.